R/mergeTreeSE.R

```diff
@@ -0,0 +1,438 @@
+#' Merge TreeSE objects into single TreeSE object.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object or a list of 
+#' \code{\link{SummarizedExperiment}} objects.
+#' 
+#' @param y a \code{\link{SummarizedExperiment}} object 
+#' 
+#' @param assay_name A single character value for selecting the
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#' to be merged. (By default: \code{assay_name = "counts"})
+#' 
+#' @param join A single character value for selecting the joining method.
+#' Must be 'full', 'inner', 'left', or 'right'. 'left' and 'right' are disabled
+#' when more than two objects are being merged.  (By default: \code{join = "full"})
+#' 
+#' @param missing_values NA, 0, or a single character values specifying the notation
+#' of missing values. (By default: \code{missing_values = 0})
+#' 
+#' @param verbose A single boolean value to choose whether to show messages. 
+#' (By default: \code{verbose = TRUE})
+#'
+#' @param ... optional arguments (not used).
+#'
+#' @return A single \code{TreeSummarizedExperiment} object.
+#'
+#' @details
+#' This function merges multiple \code{SummarizedExperiment} objects. It combines
+#' \code{rowData}, \code{assays}, and \code{colData} so that the output includes
+#' each unique row and column ones. If, for example, all rows are not shared with
+#' individual objects, there are missing values in \code{assays}. The notation of missing
+#' can be specified with the \code{missing_values} argument. 
+#' 
+#' Compared to \code{cbind} and \code{rbind} \code{mergeTreeSE} 
+#' allows more freely merging since \code{cbind} and \code{rbind} expect 
+#' that rows and columns are matching, respectively.
+#' 
+#' You can choose joining methods from 'full', 'inner',
+#' 'left', and 'right'. In all the methods, all the samples are included in the 
+#' result object. However, with different methods, it is possible to choose which 
+#' rows are included.
+#' 
+#' \itemize{
+#'   \item{\code{full} -- all unique features}
+#'   \item{\code{inner} -- all shared features}
+#'   \item{\code{left} -- all the features of the first object}
+#'   \item{\code{right} -- all the features of the second object}
+#' }
+#'
+#' @seealso
+#' \itemize{
+#'   \item{\code{TreeSummarizedExperiment::cbind}}
+#'   \item{\code{TreeSummarizedExperiment::rbind}}
+#' }
+#'
+#' @name mergeTreeSE
```

Comment:
Yes, it might. I was thinking that if we want to focus on TreeSE. However, it might be better to use more general name since the function is not limited to TreeSE

Reply:
Yes.